### PR TITLE
docs: update usage.md

### PR DESCRIPTION
**Pull request by catsmanac**
While working on multiphase feature I created a bunch of notes that I formatted and added to usage.md.

### CODE REVIEW

**Comment by catsmanac** [comment]
This last commit completes the initial setup of documentation. Next step could be add/update/improve of docstring in the source code once this skeleton is fine or altered. 

The sphinx setup now has 2 more extensions in use. No idea if that will break the CI.
 
```
extensions = [
    "myst_parser",
    "sphinx.ext.autodoc",
    "sphinx_autodoc_typehints",
]
```

afbeelding

**Comment by bdraco** [comment]
What is the intent of adding the `client` submodule?

**Comment by catsmanac** [comment]
> What is the intent of adding the `client` submodule?

My bad. That's my local test client with my envoy simulator and not part of code. Should not have been included. VSCODE is not supposed to include that in the push, but looks like I messed something up.

**Review by bdraco — APPROVED** [review]
Thanks @catsmanac

File: docs/usage_intro.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+while True:
+    data: EnvoyData = await envoy.update()
+
+    print(f'Watts: {data.system_production.watts_now}')
+    print(f'TodaysEnergy: {data.system_production.watt_hours_today}')
+    print(f'LifetimeEnergy {data.system_production.watt_hours_lifetime}')
+    print(f'Last7DaysEnergy {data.system_production.watt_hours_last_7_days}')
+
+    time.sleep(some_time)
```
Comment (bdraco) [inline docs/usage_intro.md:33]:
```suggestion
    await asyncio.sleep(some_time)
```

**Comment by bdraco** [comment]
Looks like the docs build is failing https://readthedocs.org/projects/pyenphase/builds/22523601/

I fixed the python version

```
Running Sphinx v7.2.6

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyenphase/envs/latest/lib/python3.10/site-packages/sphinx/registry.py", line 447, in load_extension
    mod = import_module(extname)
  File "/home/docs/.asdf/installs/python/3.10.13/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'sphinx_autodoc_typehints'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyenphase/envs/latest/lib/python3.10/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyenphase/envs/latest/lib/python3.10/site-packages/sphinx/application.py", line 233, in __init__
    self.setup_extension(extension)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyenphase/envs/latest/lib/python3.10/site-packages/sphinx/application.py", line 406, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/pyenphase/envs/latest/lib/python3.10/site-packages/sphinx/registry.py", line 450, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname,
sphinx.errors.ExtensionError: Could not import extension sphinx_autodoc_typehints (exception: No module named 'sphinx_autodoc_typehints')

Extension error:
Could not import extension sphinx_autodoc_typehints (exception: No module named 'sphinx_autodoc_typehints')
```

**Comment by bdraco** [comment]
I added the missing module in https://github.com/pyenphase/pyenphase/pull/115

**Comment by catsmanac** [comment]
Thanks, local build worked, but missed that entry.

**Pull request merged**